nidn/fdtd_integration/compute_spectrum_fdtd.py

```diff
@@ -153,3 +170,28 @@ def _average_along_detector(signal):
             s[2] += p[2] / len(signal[i])
         avg[i] = s
     return avg
+
+
+def _summed_thickness_times_permittivity(thicknesses, permittivity):
+    summed_permittivity = 0
+    for i in range(len(thicknesses)):
+        summed_permittivity += thicknesses[i] * permittivity[:, :, i].real
```

Comment:
shouldn't this be frequency dependent as well? permittivity isn't constant for us after all?

Reply:
maybe we take the max permittivity?

Reply:
This is frequency dependent, as this is done separate for each frequency

Reply:
Max permittivity of enitire simulation, or of the material stack?

Reply:
Ah right, I misunderstood. However, I think you need to do (permittivity[:, :, i].real).max() or something like that for the patterned layer case? permittivity.shape could in the future be other than 1 x 1 x N_layers?

Reply:
Yeah, if a patterned layer is to be implemented, the max is probably the best idea. Can do that all the time, as this will give a higher recomended number of timesteps, which is safer